core/src/main/java/de/jplag/reporting/jsonfactory/DirectoryManager.java

```diff
@@ -92,4 +116,26 @@ public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOExce
         logger.info("Display the results with the report viewer at https://jplag.github.io/JPlag/");
         return true;
     }
+
+    /**
+     * Processes file path.
+     * @param filePath The path of the file that needs to be compared
+     * @param name The name of the new directory. According to this name we can get sub-folder's structure after this
+     * directory.
+     * @return The index of the new directory in filePath
+     */
+    public static int findLastDirectory(String filePath, String name) {
+        String filePathCopy = filePath;
+        int index;
+        while (true) {
+            index = filePathCopy.lastIndexOf(name);
+            if (index == -1)
+                return filePathCopy.length();
+            String tempPath = filePathCopy.substring(0, index + name.length());
+            if (Files.isDirectory(Path.of(tempPath))) {
+                return index;
+            }
+            filePathCopy = filePathCopy.substring(0, index);
+        }
```

Comment:
At a quick glance, I cannot say what you are doing here, but it seems overcomplicated. Avoid `while(true)`, please.